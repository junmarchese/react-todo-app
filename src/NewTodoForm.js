import React, { useState } from "react";

function NewTodoForm({ addTodo }) {
    const [task, setTask] = useState("");

    const handleSubmit = (e) => {
        e.preventDefault();
        if (task.trim() === "") return;
        addTodo({ id: Date.now(), task });
        setTask("");
    };

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="task"><strong>New Todo:</strong></label>
            <input
                id="task"
                name="task"
                value={task}
                onChange={(e) => setTask(e.target.value)}
            />
            <button>Add Todo</button>
        </form>
    );
}

export default NewTodoForm;