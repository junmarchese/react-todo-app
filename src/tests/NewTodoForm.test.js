import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import NewTodoForm from "../NewTodoForm";

it("renders without crashing", () => {
    render(<NewTodoForm addTodo={() => {}} />);
});

it("matches snapshot", () => {
    const { asFragment } = render(<NewTodoForm addTodo={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
});

it("can submit a new todo", async () => {
    const mockAddTodo = jest.fn();
    render(<NewTodoForm addTodo={mockAddTodo} />);

    const input = screen.getByLabelText("New Todo:");
    const button = screen.getByText("Add Todo");

    // Fill out and submit the form
    await userEvent.type(input, "Test Todo");
    await userEvent.click(button);

    expect(mockAddTodo).toHaveBeenCalledTimes(1);
    expect(mockAddTodo).toHaveBeenCalledWith(expect.objectContaining({ task: "Test Todo" }));
    expect(input).toHaveValue("");
});



