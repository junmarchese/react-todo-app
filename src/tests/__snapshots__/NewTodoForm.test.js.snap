// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      <strong>
        New Todo:
      </strong>
    </label>
    <input
      id="task"
      name="task"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
